content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx

```diff
@@ -0,0 +1,258 @@
+---
+title: 'Referential actions'
+metaTitle: 'Referential actions (Preview)'
+metaDescription: 'Referential actions let you define the update and delete behavior of related models on the database level'
+tocDepth: 2
+preview: true
+---
+
+<TopBlock>
+
+In 2.26.0 and later, you can define referential actions on your relation fields. Referential actions determine what should happen to a record when a related record is deleted or updated.
+
+<Admonition type="warning">
+
+**If you have a project that is running version 2.25.0 and you then upgrade to 2.26.0 and enable the preview feature, using the `delete()` or `deleteAll()` methods may results in deletion of data where once it would not have.**
+
+</Admonition>
+
+When you enable the `referentialActions` preview flag, you must choose one of three upgrade paths. See the [Upgrading to referential actions guides](../../../../guides/upgrade-guides/upgrading-to-referential-actions) for more information on adopting the feature in your project
+
+</TopBlock>
+
+## What are referential actions?
+
+Referential actions are constraints that define how a referenced record is handled by the database when you run an [`update`](../../prisma-client/crud#update) or [`delete`](../../prisma-client/crud#delete) query.
+
+Prisma supports the following referential actions:
+
+- [`Cascade`](#cascade)
+- [`Restrict`](#restrict)
+- [`NoAction`](#noaction)
+- [`SetNull`](#setnull)
+- [`SetDefault`](#setdefault)
+
+The following example shows `Post` and `User` models with a [one-to-many](./#one-to-many-relations) relationship. The `Post` model has an `authorId` field which is used as the relation scalar for the `author` field, which itself points to the `User` model. 
+
+It's defined using the [`@relation`](../../../../reference/api-reference/prisma-schema-reference#relation) attribute. Under the hood this is represented as a foreign key constraint on the database table.
+
+```prisma file=schema.prisma
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id])
+  authorId  Int
+}
+
+model User {
+  id    Int         @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+The foreign key ensures that when saving any posts, the value of the `authorId` [scalar field](../data-model#scalar-fields) must be referencing an existing  `id` of the `User` model. 
+
+When a [migration](../../prisma-migrate) is run the following SQL is produced:
```

Comment:
The SQL is probably different for different databases, so we should probably mention which one this example is for here.